controls/raddiagram/styles-and-templates/styling-shapes-with-graphsource.md

```diff
@@ -0,0 +1,219 @@
+
+* __TextTool__: Creates a __RadDiagramTextShape__
+* __PathTool__: Creates a __RadDiagramShape__
+* __PencilTool__: Creates a __RadDiagramShape__
+* __ConnectorTool:__ Creates a __RadDiagramConnection__
+	
+To overcome this you can create a StyleSelector and apply it to the __ShapeStyleSelector__ of the RadDiagram. __Example 4__ demonstrates the custom __StyleSelector__.
+
+#### __[C#] Example 4: Creating a custom StyleSelector__
+	{{region raddiagram-styling-shapes-with-graphsource_3}}
+		public class ShapeStyleSelector : StyleSelector
+		{
+			public Style ContainerStyle { get; set; }
+			public Style ShapeStyle { get; set; }
+			public Style TextToolShape { get; set; }
+			public override Style SelectStyle(object item, DependencyObject container)
+			{
+				if (container is RadDiagramContainerShape)
+				{
+					return ContainerStyle;
+				}
+
+				if (container is RadDiagramShape)
+				{
+					return ShapeStyle;
+				}
+				
+				if(container is RadDiagramTextShape)
+				{
+					return TextToolShape;
+				}
+				return base.SelectStyle(item, container);
+			}
+		}
+	{{endregion}}
+
+Now you need to specify different styles for the different types of shapes. For example, when creating a shape with the TextTool, the TargetType property of the custom style needs to be set to __RadDiagramTextShape__. __Example 4__ shows the final XAML code.  
+
+#### __[XAML] Example 5: Creating custom styles for every shape__
+	{{region raddiagram-styling-shapes-with-graphsource_4}}
+		<Window.Resources>
+			<DataTemplate x:Key="contentTemplate">
+				<TextBlock Text="{Binding Content}" />
+			</DataTemplate>
+			<DataTemplate x:Key="editTemplate">
+				<TextBox Text="{Binding Content, Mode=TwoWay}" />
+			</DataTemplate>
+			<local:ShapeStyleSelector x:Key="shapeStyleSelector" >
+				<local:ShapeStyleSelector.ContainerStyle>
+					<Style TargetType="telerik:RadDiagramContainerShape">
+						<Setter Property="Position" Value="{Binding Position, Mode=TwoWay}" />
+						<Setter Property="ContentTemplate" Value="{StaticResource contentTemplate}" />
+						<Setter Property="EditTemplate" Value="{StaticResource editTemplate}" />
+					</Style>
+				</local:ShapeStyleSelector.ContainerStyle>
+				<local:ShapeStyleSelector.ShapeStyle>
+					<Style TargetType="telerik:RadDiagramShape">
+						<Setter Property="Position" Value="{Binding Position, Mode=TwoWay}" />
+						<Setter Property="ContentTemplate" Value="{StaticResource contentTemplate}" />
+						<Setter Property="EditTemplate" Value="{StaticResource editTemplate}" />
+					</Style>
+				</local:ShapeStyleSelector.ShapeStyle>
+				<local:ShapeStyleSelector.TextToolShape>
+					<Style TargetType="telerik:RadDiagramTextShape">
+						<Setter Property="Position" Value="{Binding Position, Mode=TwoWay}" />
+						<Setter Property="ContentTemplate" Value="{StaticResource contentTemplate}"/>
+						<Setter Property="EditTemplate" Value="{StaticResource editTemplate}"/>
+					</Style>
```

Comment:
We should also probably list the required BasedOn attributes when using the implicit styles theming approach.